src/js/modules/FormFeedback.js

```diff
@@ -0,0 +1,231 @@
+      }) );
+
+    this.setState({
+      step: 2,
+      emoji: e.currentTarget.value
+    });
+  }
+
+  handleTextAreaChange = (e) => {
+    this.setState({
+      feedback: e.currentTarget.value
+    });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+
+    this.logEvent('send-feedback-click', JSON.stringify({
+      'emoji': this.state.emoji,
+      'feedback': this.state.feedback
+    }));
+
+    if(!this.state.feedback) {
+      this.setState({
+        step: 3,
+        emoji: null,
+        feedback: null,
+        error: null
+      });
+      return;
+    }
+
+    this.setState({
+      loading: true,
+    })
+
+    postFeedback({
+      title: 'Alpha Site Feedback',
+      description: `**Emoji rating:**\n${this.state.emoji}\n\n**Text feedback:**\n${this.state.feedback}\n\n`,
+    })
+    .then(({data}) => {
+      this.setState({
+        step: 3,
+        loading: false,
+        emoji: null,
+        feedback: null,
+        error: null
+      })
+    })
+    .catch((e) => {
+      this.logEvent('send-feedback-error');
+      //TODO: better handle error messaging
+      console.log('ERROR:', e);
+
+      this.setState({
+        loading: false,
+        error: true
+      })
+    })
+  }
+
+  handleReset = (e) => {
+    this.logEvent('reset');
+    this.setState({
+      step: 1,
+      loading: false,
+      emoji: null,
+      feedback: null,
+      error: null
+    })
+  }
+
+  render() {
+
+    const {intl} = this.props;
+
+    return (
+      <div className="coa-FormFeedback">
+
+      { this.state.step === 1 && (
+        <div className="coa-FormFeedback__step1">
+          <SectionHeaderSerif title={intl.formatMessage(i18nMessages.step1title)} />
+          <div className="coa-FormFeedback__emojis">
+            <button onClick={this.handleEmojiClick}
+              value="-2"
+              aria-label={intl.formatMessage(i18nMessages.emojiDisappointed)}
+            >
+              <img src={emojiDisappointed} alt={intl.formatMessage(i18nMessages.emojiDisappointed)} />
+            </button>
+            <button onClick={this.handleEmojiClick}
+              value="-1"
+              aria-label={intl.formatMessage(i18nMessages.emojiSad)}
+            >
+              <img src={emojiSad} alt={intl.formatMessage(i18nMessages.emojiSad)} />
+            </button>
+            <button onClick={this.handleEmojiClick}
+              value="0"
+              aria-label={intl.formatMessage(i18nMessages.emojiNeutral)}
+            >
+              <img src={emojiNeutral} alt={intl.formatMessage(i18nMessages.emojiNeutral)} />
+            </button>
+            <button onClick={this.handleEmojiClick}
+              value="1"
+              aria-label={intl.formatMessage(i18nMessages.emojiSlightlySmiling)}
+            >
+              <img src={emojiSlightlySmiling} alt={intl.formatMessage(i18nMessages.emojiSlightlySmiling)} />
+            </button>
+            <button onClick={this.handleEmojiClick}
```

Comment:
Lots of repetition around this button. Wondering if thats a clue that we should make an `<EmojiButton />` stateless functional component to pass an emoji object as a prop. Where that object has keys like `img`, `altText`, and `value`. Doesn't need its own file imo.

Reply:
Updated this to be a map function over emoji keys. Hopefully this handles the repetition well enough